src/main/java/Rabbit/util/Parser.java

```diff
@@ -0,0 +1,224 @@
+            // if input is delete + a non-integer,
+            // throws an exception due to incorrect format
+            throw new AddToListException(AddToListException.Type.FORMAT);
+        } catch (DateTimeException e) {
+            throw new AddToListException(AddToListException.Type.TIMEFORMAT);
+        }
+    }
+
+    /**
+     * Returns the index that separates the content
+     * from the time of the task when the user creates a task.
+     *
+     * @param input the string after the task type in the input.
+     * @return the index of the character in the string before which.
+     * is the content, after which is the time, -1 if the format is wrong.
+     */
+    public static int parseContent(String input) {
+        for (int i = 0; i < input.length() - 1; i++) {
+            if (input.charAt(i) == ' ' && input.charAt(i + 1) == '/') {
+                return i + 1;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Returning the first word of the string.
+     *
+     * @param input the user's input
+     * @return the index of the character in the string
+     * before which is the first word
+     */
+    public static int parseFunction(String input) {
+        for (int i = 0; i < input.length(); i++) {
+            if (input.charAt(i) == ' ') {
+                return i + 1;
+            }
+        }
+        return input.length();
+    }
+
+    /**
+     * Returns the tasks matching the keyword of find.
+     *
+     * @param input the user's input
+     * @param list the list.
+     * @return the matching tasks.
+     * @throws FindFormatException
+     */
+    public static String parseFind(String input, TaskList list) throws FindFormatException {
+        try {
+            String content = input.substring(5);
+            if (content.length() == 0) {
+                throw new FindFormatException();
+            }
+            return content;
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new FindFormatException();
+        }
+    }
+
+    /**
+     * Parses the user's input to mark command.
+     *
+     * @param input the user's command.
+     * @param list the list of tasks.
+     * @return the index of the task.
+     * @throws MarkUnmarkException
+     */
+    public static int parseMark(String input, TaskList list) throws MarkUnmarkException {
+        try {
+            Integer.parseInt(input.substring(5));
+        } catch (NumberFormatException ex) {
+            // if input is mark + a non-integer,
+            // throws an exception due to incorrect format
+            throw new MarkUnmarkException(MarkUnmarkException.Type.MARKFORMAT);
+        }
+
+        int i = Integer.parseInt(input.substring(5));
+
+        if (i > list.size() || i <= 0) {
+            throw new MarkUnmarkException(MarkUnmarkException.Type.INDEX);
+        }
+
+        if (list.get(i - 1).isDone()) {
+            throw new MarkUnmarkException(MarkUnmarkException.Type.MARKREPEAT);
```

Comment:
I find this really well done! I feel that this exception is not one that we commonly think of so adding it was very important and well thought-out!